bindings/node.js/kzg.cxx

```diff
@@ -296,73 +274,58 @@ Napi::Value VerifyBlobKzgProof(const Napi::CallbackInfo& info) {
 
 // verifyBlobKzgProofBatch: (blobs: Blob[], commitmentsBytes: Bytes48[], proofsBytes: Bytes48[], setupHandle: SetupHandle) => boolean;
 Napi::Value VerifyBlobKzgProofBatch(const Napi::CallbackInfo& info) {
-  auto env = info.Env();
-
+  Napi::Env env = info.Env();
   C_KZG_RET ret;
-  Blob *blobs = NULL;
-  Bytes48 *commitments = NULL;
-  Bytes48 *proofs = NULL;
-  Napi::Value result = env.Null();
-
-  size_t argument_count = info.Length();
-  size_t expected_argument_count = 4;
-  if (argument_count != expected_argument_count) {
-    return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
+  Napi::Value result = env.Undefined();
+  if (!(info[0].IsArray() && info[1].IsArray() && info[2].IsArray())) {
+    Napi::Error::New(env, "blobs, commitments, and proofs must all be arrays").ThrowAsJavaScriptException();
+    return result;
   }
-
-  auto blobs_param = info[0].As<Napi::Array>();
-  auto commitments_param = info[1].As<Napi::Array>();
-  auto proofs_param = info[2].As<Napi::Array>();
+  Napi::Array blobs_param = info[0].As<Napi::Array>();
+  Napi::Array commitments_param = info[1].As<Napi::Array>();
+  Napi::Array proofs_param = info[2].As<Napi::Array>();
   auto kzg_settings = info[3].As<Napi::External<KZGSettings>>().Data();
-
-  auto blobs_count = blobs_param.Length();
-  auto commitments_count = commitments_param.Length();
-  auto proofs_count = proofs_param.Length();
-
-  if (blobs_count != commitments_count || blobs_count != proofs_count) {
-    Napi::Error::New(env, "verifyBlobKzgProofBatch requires equal number of blobs/commitments/proofs")
-      .ThrowAsJavaScriptException();
-    result = env.Null();
-    goto out;
+  uint32_t count = blobs_param.Length();
+  if (count != commitments_param.Length() || count != proofs_param.Length()) {
+    Napi::Error::New(env, "requires equal number of blobs/commitments/proofs").ThrowAsJavaScriptException();
+    return result;
   }
+  // allocate all three at once. if any fails goto out; will call free on all three so all must exist
```

Comment:
I think the previous approach of initializing these pointers on the top as NULL:
```
  Blob *blobs = NULL;
  Bytes48 *commitments = NULL;
  Bytes48 *proofs = NULL;
```

is a more standard C approach, since it ensures that even if something fails (or the code changes) and we go to `out` sooner, they will always be NULL and `free` will handle them correctly.

I also think that allocating a single array and then checking it, instead of allocating them all at once and then checking in the end is more standard, but no strong opinion here.

Reply:
AHHHH... That was my one for the day!  I updated that so it follows standard C practice.